pkg/controllers/v1alpha1/fluidapp/dataflowaffinity/dataflowaffinity_controller.go

```diff
@@ -0,0 +1,198 @@
+/*
+ Copyright 2024 The Fluid Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package dataflowaffinity
+
+import (
+	"context"
+	"fmt"
+	"github.com/fluid-cloudnative/fluid/pkg/common"
+	"github.com/fluid-cloudnative/fluid/pkg/utils"
+	"github.com/go-logr/logr"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/fluid-cloudnative/fluid/pkg/ctrl/watch"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/kubeclient"
+)
+
+const DataOpJobControllerName string = "DataOpJobController"
+
+type DataOpJobReconciler struct {
+	client.Client
+	Recorder record.EventRecorder
+	Log      logr.Logger
+}
+
+func (f *DataOpJobReconciler) ControllerName() string {
+	return DataOpJobControllerName
+}
+
+func (f *DataOpJobReconciler) ManagedResource() client.Object {
+	return &batchv1.Job{}
+}
+
+type reconcileRequestContext struct {
+	context.Context
+	Log logr.Logger
+	job *batchv1.Job
+	types.NamespacedName
+}
+
+func NewDataOpJobReconciler(client client.Client,
+	log logr.Logger,
+	recorder record.EventRecorder) *DataOpJobReconciler {
+	return &DataOpJobReconciler{
+		Client:   client,
+		Recorder: recorder,
+		Log:      log,
+	}
+}
+
+// Reconcile reconciles Jobs
+// +kubebuilder:rbac:groups=v1,resources=pods,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=v1,resources=pods/status,verbs=get;update;patch
+func (f *DataOpJobReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	requestCtx := reconcileRequestContext{
+		Context:        ctx,
+		Log:            f.Log.WithValues("namespacedName", request.NamespacedName),
+		NamespacedName: request.NamespacedName,
+	}
+	job, err := kubeclient.GetJob(f.Client, request.Name, request.Namespace)
+	if err != nil {
+		requestCtx.Log.Error(err, "fetch job error")
+		return reconcile.Result{}, err
```

Comment:
How about reusing `utils.RequeueIfError()` and `utils.NoRequeue()`?

Reply:
ok, fixed